pandas/tests/arrays/test_datetimelike.py

```diff
@@ -45,7 +45,105 @@ def datetime_index(request):
     return pi
 
 
-class TestDatetimeArray(object):
+@pytest.fixture
+def timedelta_index(request):
+    """
+    A fixture to provide TimedeltaIndex objects with different frequencies.
+     Most TimedeltaArray behavior is already tested in TimedeltaIndex tests,
+    so here we just test that the TimedeltaArray behavior matches
+    the TimedeltaIndex behavior.
+    """
+    # TODO: flesh this out
+    return pd.TimedeltaIndex(['1 Day', '3 Hours', 'NaT'])
+
+
+def index_to_array(index):
+    """
+    Helper function to construct a Datetime/Timedelta/Period Array from an
+    instance of the corresponding Index subclass.
+    """
+    if isinstance(index, pd.DatetimeIndex):
+        return DatetimeArrayMixin(index)
+    elif isinstance(index, pd.TimedeltaIndex):
+        return TimedeltaArrayMixin(index)
+    elif isinstance(index, pd.PeriodIndex):
+        return PeriodArrayMixin(index)
+    else:
+        raise TypeError(type(index))
+
+
+class SharedTests(object):
+    index_cls = None
+
+    def test_take(self):
```

Comment:
I think such basic tests are already covered in the base extension tests?

Reply:
No idea.  I'm pretty sure the dtype-specific tests must be non-redundant.